tracer/test/Datadog.Trace.Tests/Propagators/W3CBaggagePropagatorTests.cs

```diff
@@ -0,0 +1,268 @@
+    {
+        var baggage = W3CBaggagePropagator.ParseHeader(inputHeader);
+
+        if (expectedPairs is null || expectedPairs.Length == 0)
+        {
+            baggage.Should().BeNull();
+            return;
+        }
+
+        baggage.Should().NotBeNull();
+        baggage!.Count.Should().Be(expectedPairs.Length);
+
+        foreach (var pair in expectedPairs)
+        {
+            baggage[pair.Key].Should().Be(pair.Value);
+        }
+    }
+
+    [Fact]
+    public void Inject_IHeadersCollection()
+    {
+        var headers = new Mock<IHeadersCollection>();
+        var context = CreatePropagationContext();
+
+        BaggagePropagator.Inject(context, headers.Object);
+
+        headers.Verify(h => h.Set("baggage", "key1=value1,key2=value2"), Times.Once());
+        headers.VerifyNoOtherCalls();
+    }
+
+    [Fact]
+    public void Inject_CarrierAndDelegate()
+    {
+        // using IHeadersCollection for convenience, but carrier could be any type
+        var headers = new Mock<IHeadersCollection>();
+        var context = CreatePropagationContext();
+
+        BaggagePropagator.Inject(context, headers.Object, (carrier, name, value) => carrier.Set(name, value));
+
+        headers.Verify(h => h.Set("baggage", "key1=value1,key2=value2"), Times.Once());
+        headers.VerifyNoOtherCalls();
+    }
+
+    [Fact]
+    public void Extract_IHeadersCollection()
+    {
+        var headers = new Mock<IHeadersCollection>(MockBehavior.Strict);
+
+        headers.Setup(h => h.GetValues("baggage"))
+               .Returns(["key1=value1,key2=value2"]);
+
+        var result = BaggagePropagator.Extract(headers.Object);
+        headers.Verify(h => h.GetValues("baggage"), Times.Once());
+
+        var baggage = result.Baggage!;
+        baggage.Should().NotBeNull();
+        baggage["key1"].Should().Be("value1");
+        baggage["key2"].Should().Be("value2");
+
+        result.SpanContext.Should().BeNull();
+    }
+
+    [Fact]
+    public void Extract_CarrierAndDelegate()
+    {
+        // using IHeadersCollection for convenience, but carrier could be any type
+        var headers = new Mock<IHeadersCollection>(MockBehavior.Strict);
+
+        headers.Setup(h => h.GetValues("baggage"))
+               .Returns(["key1=value1,key2=value2"]);
+
+        var result = BaggagePropagator.Extract(headers.Object, (carrier, name) => carrier.GetValues(name));
+        headers.Verify(h => h.GetValues("baggage"), Times.Once());
+
+        var baggage = result.Baggage!;
+        baggage.Should().NotBeNull();
+        baggage["key1"].Should().Be("value1");
+        baggage["key2"].Should().Be("value2");
+
+        result.SpanContext.Should().BeNull();
+    }
+
+    [Theory]
+    [InlineData("foo")]
+    [InlineData("key=")]
+    [InlineData("=value")]
+    [InlineData("=")]
+    [InlineData("")]
+    [InlineData("  ")]
+    [InlineData(null)]
+    public void Extract_InvalidFormat(string header)
```

Comment:
Loving all the edge test cases here, have you considered enabling code coverage (even just locally) to see if you caught all the edge cases? It's really hard to be sure 😅

Reply:
Good idea.

Reply:
I found one case where I wasn't handling null input, and I added more test cases to get full coverage. Thanks!